e2e/testresources/kibana-legacy-monitoring.json

```diff
@@ -0,0 +1,566 @@
+{
+    "hits": {
+        "hits": [
+            {
+                "_id": "fvA-NnIBLHhfMFfAcIMH",
+                "_index": ".monitoring-kibana-7-2020.05.21",
+                "_score": null,
+                "_source": {
+                    "cluster_uuid": "RFqV_TQrR_C-tnjv-RA6yw",
+                    "interval_ms": 10000,
+                    "kibana_stats": {
+                        "concurrent_connections": 0,
+                        "kibana": {
+                            "host": "10.0.2.15",
+                            "index": ".kibana",
+                            "name": "packer-virtualbox-iso-1584284895",
+                            "snapshot": true,
+                            "status": "green",
+                            "transport_address": "10.0.2.15:5601",
+                            "uuid": "82e3cb18-da5d-41a4-ab76-0364f87df527",
+                            "version": "7.7.1-SNAPSHOT"
+                        },
+                        "os": {
+                            "distro": "Ubuntu Linux",
+                            "distroRelease": "Ubuntu Linux-16.04",
+                            "load": {
+                                "15m": 0.19189453125,
+                                "1m": 1.73583984375,
+                                "5m": 0.53515625
+                            },
+                            "memory": {
+                                "free_in_bytes": 141086720,
+                                "total_in_bytes": 4143366144,
+                                "used_in_bytes": 4002279424
+                            },
+                            "platform": "linux",
+                            "platformRelease": "linux-4.4.0-174-generic",
+                            "uptime_in_millis": 145000
+                        },
+                        "process": {
+                            "event_loop_delay": 0.5044070000003558,
+                            "memory": {
+                                "heap": {
+                                    "size_limit": 1526909922,
+                                    "total_in_bytes": 474058752,
+                                    "used_in_bytes": 260763544
+                                },
+                                "resident_set_size_in_bytes": 586629120
+                            },
+                            "uptime_in_millis": 30432
+                        },
+                        "requests": {
+                            "disconnects": 0,
+                            "total": 0
+                        },
+                        "response_times": {
+                            "average": 0,
+                            "max": 0
+                        },
+                        "timestamp": "2020-05-21T07:59:15.934Z",
+                        "usage": {
```

Comment:
@ycombinator I see differences in the kibana parity checks for `kibana_stats`:
```
[2020-05-25T18:35:54.294Z] time="2020-05-25T18:35:49Z" level=warning msg="Metricbeat-indexed doc for type='kibana_stats' has unexpected insertion" docType=kibana_stats insertion=kibana_stats.usage.uptime.last_24_hours.hits.no_of_unique_observer_locations product=kibana

[2020-05-25T18:35:54.294Z] time="2020-05-25T18:35:49Z" level=warning msg="Metricbeat-indexed doc for type='kibana_stats' has unexpected insertion" docType=kibana_stats insertion=kibana_stats.usage.uptime.last_24_hours.hits.monitor_name_stats.max_length product=kibana

[2020-05-25T18:35:54.294Z] time="2020-05-25T18:35:49Z" level=warning msg="Metricbeat-indexed doc for type='kibana_stats' has unexpected insertion" docType=kibana_stats insertion=kibana_stats.usage.uptime.last_24_hours.hits.observer_location_name_stats.max_length product=kibana

[2020-05-25T18:35:54.294Z] time="2020-05-25T18:35:49Z" level=warning msg="Metricbeat-indexed doc for type='kibana_stats' has unexpected insertion" docType=kibana_stats insertion=kibana_stats.usage.uptime.last_24_hours.hits.no_of_unique_monitors product=kibana

[2020-05-25T18:35:54.294Z] time="2020-05-25T18:35:49Z" level=warning msg="Metricbeat-indexed doc for type='kibana_stats' has unexpected insertion" docType=kibana_stats insertion=kibana_stats.usage.uptime.last_24_hours.hits.observer_location_name_stats.avg_length product=kibana

[2020-05-25T18:35:54.294Z] time="2020-05-25T18:35:49Z" level=warning msg="Metricbeat-indexed doc for type='kibana_stats' has unexpected insertion" docType=kibana_stats insertion=kibana_stats.usage.uptime.last_24_hours.hits.monitor_name_stats.min_length product=kibana

[2020-05-25T18:35:54.295Z] time="2020-05-25T18:35:49Z" level=warning msg="Metricbeat-indexed doc for type='kibana_stats' has unexpected insertion" docType=kibana_stats insertion=kibana_stats.usage.uptime.last_24_hours.hits.observer_location_name_stats.min_length product=kibana

[2020-05-25T18:35:54.295Z] time="2020-05-25T18:35:49Z" level=warning msg="Metricbeat-indexed doc for type='kibana_stats' has unexpected insertion" docType=kibana_stats insertion=kibana_stats.usage.uptime.last_24_hours.hits.monitor_name_stats.avg_length product=kibana

[2020-05-25T18:35:54.295Z] time="2020-05-25T18:35:49Z" level=warning msg="Metricbeat-indexed doc for type='kibana_stats' has unexpected insertion" docType=kibana_stats insertion=kibana_stats.usage.uptime.last_24_hours.hits.autoRefreshEnabled product=kibana

[2020-05-25T18:35:54.295Z] time="2020-05-25T18:35:49Z" level=info msg="Expected parity between Metricbeat-indexed doc and legacy-indexed doc found." 
```
Please see here: https://apm-ci.elastic.co/blue/organizations/jenkins/stack%2Fe2e-testing-mbp/detail/PR-113/28/pipeline#step-296-log-519

Reply:
>  let's just focus on one product: Elasticsearch for now

>> Nevertheless, I agree that we can add them in a follow-up PR.

Again, let's please remove Kibana and other products (except ES) from this PR. Let's tighten the scope of this PR so we can get the ES stack monitoring parity tests ported over, passing, and this PR merged. Then we can add the other products' tests in follow up PRs.

Reply:
I've removed any other product apart from Elasticsearch (I guess we will simply revert that commit in a very short time 😄)

Besides that, I found 5 new insertions and 2 deletions, which I added in a separate commit (https://github.com/elastic/e2e-testing/pull/113/commits/94ccffd43dcff69ab0b945a914463ee28407dfe2), so you can investigate where they come from.

Now the ES tests are passing 🎉 

I'm happy we did this, as I was able to refactor the code a bit and also add unit tests using the JSON files from the existing Ansible tests. Those unit tests are executing all functions handling special cases, creating a safety net for future refactors.

Please let me know if I can help with the review, and thanks for your guidance!